### PR TITLE
Check for invalid PSS params before writing and use fixed width integer type in DER writer

### PR DESCRIPTION
**Pull request by faramir-dev**
##### Checklist

- [x] documentation is added or updated (internal doc only)
- [x] tests are added or updated

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

File: providers/common/der/der_rsa_key.c [hunk 1]
```diff
@@ -338,8 +347,8 @@ int ossl_DER_w_RSASSA_PSS_params(WPACKET *pkt, int tag,
 
     return ossl_DER_w_begin_sequence(pkt, tag)
         && (trailerfield == default_trailerfield
-            || ossl_DER_w_ulong(pkt, 3, trailerfield))
-        && (saltlen == default_saltlen || ossl_DER_w_ulong(pkt, 2, saltlen))
+            || ossl_DER_w_uint32(pkt, 3, trailerfield))
+        && (saltlen == default_saltlen || ossl_DER_w_uint32(pkt, 2, saltlen))
```
Comment (mattcaswell) [inline providers/common/der/der_rsa_key.c:351]: There is an implicit cast here from `int` to `uint32_t` (with both saltlen and trailerfield). Do we ever run on a platform where `INT_MAX > UINT32_MAX`? Perhaps there should be an explicit check for this?

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
@paulidale still  OK?

**Review by mattcaswell — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you.

**Comment by paulidale** [comment]
This fix is broken.  `(unsigned int)saltlen` **must** always be ≤ `UINT32_MAX`.  It's not at all clear what this check is for, there is no possibility of overflow when casting since negative values are already covered.  I suggest raising a new PR with it removed.

**Comment by paulidale** [comment]
Coverity found it: CID 1506703

**Pull request closed without merging**